### PR TITLE
Convert label to typescript

### PR DESCRIPTION
**Pull request by bautistaaa**
Closes https://github.com/mondaycom/monday-ui-react-core/issues/912

Converts `label` to typescript

#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [ ] PR has description.
- [ ] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [ ] Stories include all flows of using the component.
#### Tests
- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

File: src/components/Label/__tests__/__snapshots__/label-snapshot-tests.jest.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,196 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (laviomri) [inline src/components/Label/__tests__/__snapshots__/label-snapshot-tests.jest.tsx.snap:1]: Please remove the matching `jest.jsx.snap` file, now that it's redundant

File: src/components/Label/Label.tsx [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
-  color,
+  kind = "fill",
+  color = "primary",
   text = "",
-  isAnimationDisabled,
-  isLegIncluded
-}) => {
+  isAnimationDisabled = false,
+  isLegIncluded = false
+}: Partial<LabelProps>) => {
```
Comment (laviomri) [inline src/components/Label/Label.tsx:37]: I think it's best to mark the properties of `LabelProps` as optionals, and use `LabelProps` directly here (without wrapping it with `Partial`). This way, if there will be a required prop (from some reason), we'll only need to update `LabelProps` in a single place.

File: src/components/Label/Label.tsx [hunk 3]
```diff
@@ -1,21 +1,40 @@
...
+}
 
 const Label = ({
-  // Backward compatibility for enum naming
   className,
   wrapperClassName,
-  kind,
-  color,
+  kind = "fill",
+  color = "primary",
```
Comment (laviomri) [inline src/components/Label/Label.tsx:33]: Please use the values from the consts
```suggestion
  kind = LABEL_TYPES.FILL,
  color = LABEL_COLORS.PRIMARY,
```

**Comment by laviomri** [comment]
Hey @bautistaaa , thank you for this PR! I left you some comments. Please take a look at them, and mention @orrgottlieb once fixed. (I will be unavailable for a few days).
Good luck! 😃

**Comment by bautistaaa** [comment]
@orrgottlieb addressed all the feedback. LMK if you see anything suspicious

**Comment by orrgottlieb** [comment]
@bautistaaa going over this again. 
please comment in the issue so i could assign the issue to you 🙏

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**